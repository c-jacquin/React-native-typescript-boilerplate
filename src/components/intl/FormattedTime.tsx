import React, { StatelessComponent } from 'react'
import { FormattedTime } from 'react-intl'
import { Text } from 'react-native'

import { IFormattedTime } from './types'

const NativeFormattedTime: StatelessComponent<IFormattedTime> = props => {
    return (
        <FormattedTime {...props}>
            {(localized: string) =>
                <Text style={props.style}>
                    {localized}
                </Text>}
        </FormattedTime>
    )
}

export default NativeFormattedTime
