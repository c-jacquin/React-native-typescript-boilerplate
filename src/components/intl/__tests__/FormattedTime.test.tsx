import * as React from 'react'
import { Text } from 'react-native'
import * as renderer from 'react-test-renderer'
import { IntlProvider } from 'react-intl'

import FormattedTime from '../FormattedTime'

describe('FormattedTime', () => {
    it('should renders correctly', () => {
        const tree = renderer
            .create(
                <IntlProvider locale="en">
                    <FormattedTime style={{}} value={1500} />
                </IntlProvider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
