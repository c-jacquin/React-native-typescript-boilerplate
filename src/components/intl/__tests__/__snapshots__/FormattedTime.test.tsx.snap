// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormattedTime should renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}
>
  1:00 AM
</Text>
`;
